packages/material-ui-utils/src/deepmerge.js

```diff
@@ -2,6 +2,10 @@ export function isObject(item) {
   return item && typeof item === 'object' && !Array.isArray(item);
 }
 
+function isElement(element) {
+  return element instanceof Element || element instanceof HTMLDocument;
```

Comment:
Rule of thumb: Allowlists over disallowlists i.e. only allow plain objects. Otherwise we'll regress in other places as well.

Reply:
Good point. What do you thin of replacing `isObject` by this function:
```js
export function isPlainObject(item) {
  return item && typeof item === 'object' && item.constructor === Object;
}
```

(from https://stackoverflow.com/a/5878101/1713469)

Or would you prefer using something like https://www.npmjs.com/package/is-plain-object ?